pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -49,10 +52,18 @@ func NewPgxIngestor(conn pgxconn.PgxConn) (*DBIngestor, error) {
 //     tts the []Timeseries to insert
 //     req the WriteRequest backing tts. It will be added to our WriteRequest
 //         pool when it is no longer needed.
-func (i *DBIngestor) Ingest(tts []prompb.TimeSeries, req *prompb.WriteRequest) (uint64, error) {
-	data, totalRows, err := i.parseData(tts, req)
+func (i *DBIngestor) Ingest(tts []prompb.TimeSeries, req *prompb.WriteRequest, ha *ha.HA) (uint64, error) {
```

Comment:
since the ha state is not modified throughout the lifetime of the ingestor, we're always working with the same instance, it's better to store it as a unexported field of DBIngestor, no need to pass it around

Reply:
would also remove the requirement to modify all the tests that call Ingest